src/player/Player.php

```diff
@@ -649,6 +656,14 @@ public function getLocale() : string{
 		return $this->locale;
 	}
 
+	/**
+	 * @return Language
+	 */
+	public function getLanguage(): Language
```

Comment:
Are you sure it should return a `Language` instance instead of a `string`? `Language` currently implies a set of mapping from translation keys to translation values rather than a "language" in its intuitive definition. Is it really ideal to require `CommandSender` to return a set of translations rather than just its own preferred language type?

Reply:
> Are you sure it should return a `Language` instance instead of a `string`? `Language` currently implies a set of mapping from translation keys to translation values rather than a "language" in its intuitive definition. Is it really ideal to require `CommandSender` to return a set of translations rather than just its own preferred language type?

Why is this logic not done in `Server`?

Reply:
That's exactly what I'm saying. `CommandSender` should just return a locale name and let server resolve the locale name into a `Language` object.

Reply:
> That's exactly what I'm saying. `CommandSender` should just return a locale name and let server resolve the locale name into a `Language` object.

I don't understand why this is, if you can just take `Language` from `Server` and `Player`

Reply:
This requires `Player` implementations to manage a reference to the global translation data, or otherwise involves a dependency on `Server` in `Player::getLanguage()`.

In particular, if this method is moved to `CommandSender`, It is an anti-pattern to expect an interface method *always* be implemented by delegating to another function (in particular, a stateful one that depends on irrelevant parts, most likely undefined, of the implementation), while the method could be alternatively defined to return the sole differentiating data such that the caller calls such delegation on their own (in particular using their own state if the delegation is stateful).

Reply:
> This requires `Player` implementations to manage a reference to the global translation data, or otherwise involves a dependency on `Server` in `Player::getLanguage()`.
> 
> In particular, if this method is moved to `CommandSender`, It is an anti-pattern to expect an interface method _always_ be implemented by delegating to another function (in particular, a stateful one that depends on irrelevant parts, most likely undefined, of the implementation), while the method could be alternatively defined to return the sole differentiating data such that the caller calls such delegation on their own (in particular using their own state if the delegation is stateful).

I think we should do this: https://github.com/lunaste/PocketMine-MP/commit/13d1775f129f9f0bd37d7f239ca906439abcfdac, the `Player class` does not clog this PR from the word, because only one variable was added, which is a link